secure/push-metrics/src/lib.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+mod counters;
+mod pusher;
+
+pub use crate::{
+    counters::{Counter, Gauge, Metrics, PushMetricFormat},
+    pusher::MetricsPusher,
+};
+
+/// Defines a list of counters with help messages
+/// Additionally, requires a prefix which will be prepended to all counters
+///
+/// # Example of defining two counters and using the prefix libra_safety_rules_ for counter names
+///
+/// ```ignore
+/// define_counters![
+///     "libra_safety_rules",
+///     (some_counter: Counter, "some_counter help description"),
+///     (some_gauge: Gauge, "some_gauge help description"),
+/// ];
+/// ```
```

Comment:
should have a more complete doc block showing how to use / register etc the counters.

Reply:
Done